drivers/include/bq2429x.h

```diff
@@ -0,0 +1,500 @@
+    BQ2429X_VREG_4400   = 56,           /**< 4400 mV */
+} bq2429x_charge_voltage_limit_t;
+
+/**
+ * @brief   BQ2429x device parameters
+ */
+typedef struct {
+    i2c_t i2c;                              /**< I2C device */
+#if IS_USED(MODULE_BQ2429X_INT)
+    gpio_t int_pin;                         /**< Interrupt pin */
+#endif
+    gpio_t ce_pin;                          /**< Charge Enable pin (optional) */
+    gpio_t otg_pin;                         /**< OTG Enable pin (optional) */
+    bq2429x_input_voltage_limit_t vlim;     /**< Voltage limit */
+    bq2429x_input_current_limit_t ilim;     /**< Current limit */
+    bq2429x_charge_current_t ichg;          /**< Charge current limit */
+    bq2429x_charge_voltage_limit_t vreg;    /**< Charge voltage limit */
+} bq2429x_params_t;
+
+/**
+ * @brief   BQ2429x device descriptor
+ */
+typedef struct {
+    bq2429x_params_t params;            /**< Device parameters */
+} bq2429x_t;
+
+/**
+ * @brief   Initialize device
+ *
+ * @pre @p dev != NULL && @p params != NULL
+ *
+ * @param[out]  dev     Device descriptor.
+ * @param[in]   params  Device parameters.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ * @return BQ2429X_ERR_GPIO on GPIO initialization failure (CE or OTG pin
+ *         failed).
+ */
+int bq2429x_init(bq2429x_t *dev, const bq2429x_params_t *params);
+
+#if IS_USED(MODULE_BQ2429X_INT) || DOXYGEN
+/**
+ * @brief   Callback function for BQ2429x interrupts.
+ */
+typedef void (* bq2429x_int_cb_t)(void *);
+
+/**
+ * @brief   Initialize interrupt support for the device.
+ *
+ * @pre @p dev != NULL && @p cb != NULL
+ *
+ * The callback @p cb is called in an ISR context, so keep in mind that heavy
+ * work shouldn't be done there.
+ *
+ * @note @ref bq2429x_init MUST have been called before!
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[in]   cb      Callback called on interrupt.
+ * @param[in]   arg     Argument to be passed when the callback is called.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_GPIO on GPIO initialization failure.
+ */
+int bq2429x_init_int(bq2429x_t *dev, bq2429x_int_cb_t cb, void *arg);
+#endif /* IS_USED(MODULE_BQ2429X_INT) || DOXYGEN */
+
+/**
+ * @brief   Get device status.
+ *
+ * @pre @p dev != NULL && @p status != NULL
+ *
+ * @param[in]   dev         Device descriptor.
+ * @param[out]  status      Pointer where device status will be stored.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C error.
+ */
+int bq2429x_get_status(const bq2429x_t *dev, bq2429x_status_t *status);
+
+/**
+ * @brief   Get device faults.
+ *
+ * @pre @p dev != NULL && @p fault != NULL
+ *
+ * @param[in]   dev         Device descriptor.
+ * @param[out]  fault       Pointer where device faults will be stored.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C error.
+ */
+int bq2429x_get_fault(const bq2429x_t *dev, bq2429x_fault_t *fault);
+
+/**
+ * @brief   Enable OTG.
```

Comment:
What does it mean to enable OTG?
Charge via USB? Give charge to a USB device?

Reply:
Documented at 36e9c93cc4540733c481df1f733c2f5bb0031013